src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesWhenExpressions.java

```diff
@@ -0,0 +1,57 @@
+/*
+UnnecessaryParentheses
+tokens = (default)EXPR, IDENT, NUM_DOUBLE, NUM_FLOAT, NUM_INT, NUM_LONG, \
+         STRING_LITERAL, LITERAL_NULL, LITERAL_FALSE, LITERAL_TRUE, ASSIGN, \
+         BAND_ASSIGN, BOR_ASSIGN, BSR_ASSIGN, BXOR_ASSIGN, DIV_ASSIGN, \
+         MINUS_ASSIGN, MOD_ASSIGN, PLUS_ASSIGN, SL_ASSIGN, SR_ASSIGN, STAR_ASSIGN, \
+         LAMBDA, TEXT_BLOCK_LITERAL_BEGIN, LAND, LITERAL_INSTANCEOF, GT, LT, GE, \
+         LE, EQUAL, NOT_EQUAL, UNARY_MINUS, UNARY_PLUS, INC, DEC, LNOT, BNOT, \
+         POST_INC, POST_DEC
+
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.coding.unnecessaryparentheses;
+
+public class InputUnnecessaryParenthesesWhenExpressions {
+
+    void test(Object o) {
+
+      switch (o) {
+          case Integer i when (i > 0) -> {}  // violation 'Unnecessary paren.* around expression'
+          // violation below 'Unnecessary paren.* around expression'
+          case String s when (!s.isEmpty()) -> {}
+
+          case Point(boolean x, boolean y, boolean z)
+              // violation below 'Unnecessary paren.* around expression'
+                    when (x && y) || z -> System.out.println();
+
+          case Point p when ((p.x && p.y) || p.z) -> System.out.println();
+          // 2 violations above:
+          //                   'Unnecessary paren.* around expression'
+          //                    'Unnecessary paren.* around expression'
+
+          case Point2(int x, int y) when ((x >= 0 && y >= 0)) -> {}
+          // violation above 'Unnecessary paren.* around expression'
+
+          case Point2(int x, int y) when (x >= 5 && y >= 5) -> {}
+          // violation above 'Unnecessary paren.* around expression'
```

Comment:
Given these 2 examples and line 50. Shouldn't we have 2 violations on line 35

Reply:
Sounds like a general bug not related to `when` 

```
PS D:\CS\test> cat src/Test.java
class Test {
    void test(int x) {
        if (((x == 3))) { }
    }
}
PS D:\CS\test> java  -jar checkstyle-10.17.0-all.jar -c config.xml  src/Test.java
Starting audit...
[ERROR] D:\CS\test\src\Test.java:3:13: Unnecessary parentheses around expression. [UnnecessaryParentheses]
Audit done.
Checkstyle ends with 1 errors.

```

Reply:
Yes, please check for existing issues and if one is not present, let's open one. Please leave a comment above this one with `// until ...`, it gives me joy to see these go away in bug fix PRs.

Reply:
https://github.com/checkstyle/checkstyle/issues/15317

Reply:
done